docs/DeveloperGuide.md

```diff
@@ -1,9 +1,263 @@
+![add_png](puml_img/Add.png)
+
+> Note: P2:Project is defined as an entity that is already existing in the project list ArrayList<Project>, the position
+> of which should be lower than ArrayList<Project>. However, due to a limitation in PlantUML, the position of P3 is 
+> higher than ArrayList<Project>.
+
+**Design Consideration**
+
+Aspect: How add executes
+* Alternative 1 (current choice):
+  * The add feature will do a loop through all the projects in the ArrayList projects to see if the project has already
+    existed. For each project, it will loop through all the resources to see if a resource with the same URL has already
+    existed.
+    
+    > If the project does not exist in the ArrayList:
+    >   * Create a new project and append the resource into its resource list.
+    
+    > If the project does exist:
+    > * If resource with the same URL exists:
+    >   * Prompt the user that the resource with the same URL has already existed
+        in the project's resource list.
+    > * If resource with the same URL does not exist:
+    >   * Append the new resource into the project resource list.
+
+* Alternative 2 (none)
+---
+### List all projects & their respective resources feature
+**Proposed Implementation**  
+
+Given below is an example usage scenario and how the list-all mechanism behaves at each step.  
+
+**Step 1**:
+The user launches the application for the first time. The application’s initialised database will be empty at the start.
+
+**Step 2**:
+The user executes the following command in order to add a certain project’s resources.  
+
+>add p/CS2113 url/www.traceyourproj.com  
+
+>add p/CS3223 url/www.kaggle.com
+
+**Step 3**:
+The user wishes to see the list of resources for all projects. He executes the following command to see them.  
+
+>list-all
+
+**Design Consideration**  
+
+Aspect: How list-all executes
+* Alternative 1 (current choice):
+  * This list-all feature will do a loop through all the projects in the projects arraylist. For each project, it will loop through all the resources and print it out.
+* Alternative 2 (none)
+
+The following sequence diagram shows how the list-all operation works:
+![add_png](puml_img/list-allfinal.png)
+
+### List all the resources for a particular project feature
+**Proposed Implementation**  
+
+Given below is an example usage scenario and how the list-all mechanism behaves at each step.  
+
+
+**Step 1**:
+The user launches the application for the first time. The application’s initialised database will be empty at the start.
+
+**Step 2**:
+The user executes the following command in order to add a certain project’s resources.
+> add p/IT3011 url/www.apple.com  
+
+> add p/CZ2003 url/www.google.com  
+
+**Step 3**:
+The user wishes to see the list of resources for one project, CZ2003. He executes the following command to see them.  
+
+>list *CZ2003*
+
+**Design Consideration**  
+
+Aspect: How list PROJECTNAME executes  
```

Comment:
Can consider using proper command syntax notation (<> for required parameters, [] for optional parameters...)